networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/libp2p/LibP2PPrivateKeyLoader.java

```diff
@@ -39,12 +39,12 @@ public LibP2PPrivateKeyLoader(
   public PrivKey get() {
     final Bytes privKeyBytes =
         privateKeySource
-            .map(PrivateKeySource::getPrivateKeyBytes)
-            .orElseGet(this::generateNewPrivateKey);
+            .map(source -> source.getOrGeneratePrivateKeyBytes(keyValueStore))
+            .orElseGet(() -> generateNewPrivateKey(keyValueStore));
```

Comment:
This logic means we will never reach the `orElseGet(..)` code path, potentially losing existing functionality/behaviour from `LibP2PPrivateKeyLoader#generateNewPrivateKey(..)`.